checker/pass_context_first_checker.go

```diff
@@ -0,0 +1,67 @@
+package checker
+
+import (
+	"fmt"
+	"go/ast"
+)
+
+func init() {
+	must(Register("pass_context_first", NewPassContextFirstChecker))
+}
+
+// PassContextFirstChecker checks that if a function declaration contains a context, it is
+// the first argument.
+type PassContextFirstChecker struct{}
+
+// NewPassContextFirstChecker constructs a PassContextFirstChecker.
+func NewPassContextFirstChecker(configData interface{}) NodeChecker {
+	return &PassContextFirstChecker{}
+}
+
+// Register implements the NodeChecker interface.
+func (c *PassContextFirstChecker) Register(fc *FileChecker) {
+	fc.On(&ast.FuncDecl{}, c)
+}
+
+// Check implements the NodeChecker interface.
+func (c *PassContextFirstChecker) Check(
+	node ast.Node,
+	content string,
+	report *Report) {
+
+	decl := node.(*ast.FuncDecl)
+
+	if decl.Type.Params == nil {
+		return
+	}
+
+	contextNotFirst := false
```

Comment:
Move this to line number 43 to shorten the scope of the variable. The `len(params)` check does not require the `contextNotFirst` variable.

Reply:
Done.